client.go

```diff
@@ -80,7 +80,7 @@ func (c *Client) GetEnvironmentFlags(ctx context.Context) (f Flags, err error) {
 	if c.defaultFlagHandler != nil {
 		return Flags{defaultFlagHandler: c.defaultFlagHandler}, nil
 	}
-	return Flags{}, fmt.Errorf("%w and no default handler present", err)
+	return Flags{}, &FlagsmithAPIError{msg: fmt.Sprintf("Failed to fetch flags with error: %s", err)}
```

Comment:
I think this makes sense right, because there was an API error and no default flag handler existed so we're just continuing to pass on the API error? The question I guess is whether this could be something like a `FlagsmithClientError` because it's kind of an error in the configuration by not having a default flag handler?

Reply:
Yeah, I did spend some time thinking about it, none of them are 100 percent correct, but I think `FlagsmithClientError` is a bit more close